src/main.cpp

```diff
@@ -1221,15 +1221,6 @@ bool AcceptToMemoryPoolWorker(CTxMemPool &pool,
             CCoinsViewMemPool viewMemPool(pcoinsTip, pool);
             view.SetBackend(viewMemPool);
 
-            // do we already have it?
-            bool fHadTxInCache = pcoinsTip->HaveCoinsInCache(hash);
-            if (view.HaveCoins(hash))
-            {
-                if (!fHadTxInCache)
-                    vHashTxnToUncache.push_back(hash);
-                return state.Invalid(false, REJECT_ALREADY_KNOWN, "txn-already-known");
-            }
-
```

Comment:
I took this out because it makes no sense...why would we delete a coin from the dbcache just because we already have it there...now we have it, now we don't...furthermore, why would we reject a txn based on what the state of the dbacahe is...the coin could be there already from finishing up IBD and there are ouputs left over in the dbcache or perhaps in the future for some other reason...if we're not going to allow a txn in the mempool it has to be because it's either an invalid txn or because we already have it in the mempool...checking the dbcache is not a good or reliable test IMO for whether the txn is in the mempool.